packages/matter.js/src/behavior/system/network/NetworkServer.ts

```diff
@@ -101,21 +99,156 @@ export class NetworkServer extends NetworkBehavior {
             this.internal.runtime.endUncommissionedMode();
         }
     }
+
+    #configureListeners(config: unknown[]) {
+        const listen = Array<NetworkServer.Address>();
+        let hasUdp = false;
+        let hasBle = false;
+        let disabledBle = false;
+        for (const addr of config) {
+            if (typeof addr !== "object") {
+                throw new ImplementationError("Listen address is not an object");
+            }
+
+            let { protocol, port } = addr as Record<string, any>;
+            const { address } = addr as Record<string, any>;
+
+            if (protocol === undefined) {
+                protocol = "udp";
+            }
+
+            switch (protocol) {
+                case "ble":
+                    if (Ble.enabled) {
+                        if (hasBle) {
+                            throw new NotImplementedError("Currently only a single BLE transport is allowed");
+                        } else {
+                            hasBle = true;
+                        }
+                        if (address !== undefined) {
+                            throw new NotImplementedError("Currently you may not specify HCI ID for BLE transport");
```

Comment:
That's the goal but will require refactoring Ble.ts to convert Ble from a singleton and add HCI option.  Then we need to make it so multiple servers that share the same BLE address can share the broadcaster.  We need something similar for the MDNS side of things so figured it might make sense to do that first.

So anyway, thought this was a good intermediate place to stop.

Reply:
Maybe lets have a chat about that ... current realizy on nodejs and also on mobile devices is that there is in fact one BLE radio to use ... also bleno/noble is kind of limited to one right now and weould need a rewrite to make that configurable. Multiple servers using the same BLE radio in fact make not sense at all ... that would only be a topic when bleno would be rewritten for such a support (there was such a PR somewhen but would be need to revamped completely). So in my eyes ble can only work for one of the Server/Controllers on the Host right now.

Ok for MDNS I thought we have whats needed already, but ok maybe I miss something ... so maybe lets have a chat what you plan/imagine...